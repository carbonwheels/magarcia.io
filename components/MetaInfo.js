import Moment from 'react-moment';
import PropTypes from 'prop-types';
import css from './MetaInfo.module.css';

const MetaInfo = ({ year, month, day, readingTime, small = false }) => (
  <div className={`${css.postMetaInline} ${small ? css.small : ''}`}>
    <Moment date={`${year}-${month}-${day}`} format="ll" />
    <span className={css.middotDivider} />
    <span className={css.readingTime} title={readingTime.text} />
  </div>
);

MetaInfo.propTypes = {
  year: PropTypes.number.isRequired,
  month: PropTypes.number.isRequired,
  day: PropTypes.number.isRequired,
  readingTime: PropTypes.string.isRequired,
  small: PropTypes.bool
};

MetaInfo.defaultProps = {
  small: false
};

export default MetaInfo;
