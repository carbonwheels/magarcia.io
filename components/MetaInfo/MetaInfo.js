import Moment from 'react-moment';
import PropTypes from 'prop-types';
import css from './MetaInfo.module.css';

const MetaInfo = ({ date, readingTime, small = false }) => (
  <div className={`${css.postMetaInline} ${small ? css.small : ''}`}>
    <Moment date={date} format="ll" />
    <span className={css.middotDivider} />
    <span className={css.readingTime} title={readingTime} />
  </div>
);

MetaInfo.propTypes = {
  date: PropTypes.string.isRequired,
  readingTime: PropTypes.string.isRequired,
  small: PropTypes.bool
};

MetaInfo.defaultProps = {
  small: false
};

export default MetaInfo;
