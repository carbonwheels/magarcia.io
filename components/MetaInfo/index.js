import MetaInfo from './MetaInfo';

export default MetaInfo;
