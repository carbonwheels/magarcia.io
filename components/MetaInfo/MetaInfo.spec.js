import React from 'react';
import renderer from 'react-test-renderer';
import MetaInfo from './MetaInfo';

it('renders correctly', () => {
  const tree = renderer
    .create(<MetaInfo date="2019-13-01" readingTime="5 mins" small={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when is small', () => {
  const tree = renderer.create(<MetaInfo date="2019-13-01" readingTime="5 mins" small />).toJSON();
  expect(tree).toMatchSnapshot();
});
