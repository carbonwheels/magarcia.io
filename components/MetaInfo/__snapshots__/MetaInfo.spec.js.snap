// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="postMetaInline "
>
  <time
    dateTime={null}
  >
    Invalid date
  </time>
  <span
    className="middotDivider"
  />
  <span
    className="readingTime"
    title="5 mins"
  />
</div>
`;

exports[`renders correctly when is small 1`] = `
<div
  className="postMetaInline small"
>
  <time
    dateTime={null}
  >
    Invalid date
  </time>
  <span
    className="middotDivider"
  />
  <span
    className="readingTime"
    title="5 mins"
  />
</div>
`;
